addons/l10n_tr_nilvera_edespatch/models/stock_picking.py

```diff
@@ -0,0 +1,241 @@
+        tracking=True,
+    )
+    l10n_tr_nilvera_carrier_id = fields.Many2one(string="Carrier (TR)", comodel_name='res.partner')
+    l10n_tr_nilvera_buyer_id = fields.Many2one(
+        string="Buyer",
+        comodel_name='res.partner'
+    )
+    l10n_tr_nilvera_seller_supplier_id = fields.Many2one(
+        string="Seller Supplier",
+        comodel_name='res.partner'
+    )
+    l10n_tr_nilvera_buyer_originator_id = fields.Many2one(
+        string="Buyer Originator",
+        comodel_name='res.partner'
+    )
+    l10n_tr_nilvera_delivery_printed_number = fields.Char("Printed Delivery Note Number")
+    l10n_tr_nilvera_delivery_date = fields.Date("Printed Delivery Note Date")
+    l10n_tr_vehicle_plate = fields.Char("Vehicle Plate")
+    l10n_tr_nilvera_trailer_plate_ids = fields.Many2many(
+        string="Trailer Plates",
+        comodel_name='l10n_tr.nilvera.trailer.plate',
+        relation='l10n_tr_nilvera_delivery_vehicle_rel'
+    )
+    l10n_tr_nilvera_driver_ids = fields.Many2many(string="Drivers", comodel_name='res.partner')
+    l10n_tr_nilvera_delivery_notes = fields.Char(string="Delivery Notes")
+    l10n_tr_nilvera_despatch_state = fields.Selection(
+        string="State",
+        selection=[('to_send', "To Send"), ('sent', "Sent")],
+        compute='_compute_l10n_tr_nilvera_despatch_state',
+        store=True,
+        readonly=False,
+        tracking=True,
+    )
+    l10n_tr_nilvera_edespatch_warnings = fields.Json(copy=False)
+
+    @api.depends('company_id')
+    def _compute_l10n_tr_nilvera_despatch_scenario(self):
+        self.filtered(
+            lambda p: p.company_id.account_fiscal_country_id.code == 'TR'
+        ).l10n_tr_nilvera_despatch_scenario = 'TEMELIRSALIYE'
+ 
+    @api.depends('company_id')
+    def _compute_l10n_tr_nilvera_despatch_type(self):
+        self.filtered(
+            lambda p: p.company_id.account_fiscal_country_id.code == 'TR'
+        ).l10n_tr_nilvera_despatch_type = 'SEVK'
+
+    @api.depends('state')
+    def _compute_l10n_tr_nilvera_despatch_state(self):
+        self.filtered(
+            lambda p: p.state == 'done' and p.company_id.account_fiscal_country_id.code == 'TR'
+        ).l10n_tr_nilvera_despatch_state = 'to_send'
+
+    def _l10n_tr_validate_edelivery_fields(self):
+        self.ensure_one()
+        if self.state != 'done':
+            self.l10n_tr_nilvera_edespatch_warnings = {
+                'invalid_transfer_state': {
+                    'message': _("Please validate the transfer first to generate the XML"),
+                }
+            }
+            return
+        if self.l10n_tr_nilvera_despatch_scenario == 'HALTIPI':
+            self.l10n_tr_nilvera_edespatch_warnings = {
+                'invalid_delivery_scenario': {
+                    'message': _("Despatch Scenario as Hal Tipi is currently unsupported in the module."),
+                }
+            }
+            return
+        partners = (
+            self.env.company.partner_id
```

Comment:
Why environment `company`?
```suggestion
            self.company_id.partner_id
```